frontend/packages/yki/src/components/registration/examSession/PublicExamSessionListingFilters.tsx

```diff
@@ -28,6 +27,169 @@ import {
   selectFilteredPublicExamSessions,
 } from 'redux/selectors/examSessions';
 
+const municipalityToComboBoxOption = (m: string) => ({
+  value: m,
+  label: m,
+});
+
+const SelectMunicipality = () => {
+  const municipalities = useAppSelector(examSessionsSelector).municipalities;
+  const { municipality } = useAppSelector(examSessionsSelector).filters;
+  const dispatch = useAppDispatch();
+  const onMunicipalityChange = useCallback(
+    (municipality?: string) => {
+      dispatch(setPublicExamSessionFilters({ municipality }));
+    },
+    [dispatch]
+  );
+
+  const { t } = usePublicTranslation({
+    keyPrefix: 'yki.pages.registrationPage',
+  });
+
+  return (
+    <div className="public-exam-session-filters__filter public-exam-session-filters__municipality">
+      <Typography
+        variant="h3"
+        component="label"
+        htmlFor="public-exam-session-filters__municipality-filter"
+      >
+        {t('labels.selectMunicipality')}
+      </Typography>
+      <ComboBox
+        id="public-exam-session-filters__municipality-filter"
+        variant={TextFieldVariant.Outlined}
+        values={municipalities.map(municipalityToComboBoxOption)}
+        value={municipality ? municipalityToComboBoxOption(municipality) : null}
+        onChange={(v?: string) => {
+          const municipality = v;
+          onMunicipalityChange(municipality);
+        }}
```

Comment:
```suggestion
        onChange={onMunicipalityChange}
```

Eiks tää voi olla ihan vaan näin? Vai onko tässä joku juttu mitä en tajua. 🤔

Reply:
Joo, varmaan joku vauhtisokeus iskenyt kun olin noita massana muuttanut.